mantidproject/mantid
# Engg GUI: enable single peak fitting for consecutive run

**Description of work.**

Single peak fitting for consecutive run number's focused files has now been enabled. When a valid range of consecutive run numbers is given, the interface will automatically import and add the run number/s to the list view on the right side of the graph. The interface will then automatically update the Plot Bank combo-box according to the bank files found for each entered/selected run-number.

Also moved lot of Fitting logic related code to Presenter.

**To test:**

> Test 1

Files can be found at: `\\olympic\Babylon5\Public\Shahroz\FittingTest\`
Copy the files to focus output directory or set above directory as the focus output default directory via `settings` tab
Go to: `Interfaces/Diffraction/Engineering Diffraction/Fitting`
Enter `Focus Run #:` 194547-194551
The following run files should be found within the focused directory and imported on the right hand side of the interface within the list view widget
First run number and bank file should be selected by default
Make sure that all the banks are found within the combo-box upon selection each run number:

| Run | Bank |
| --- | --- |
| `194547` | `1,2,4,5,6` |
| `194548` | `1` |
| `194549` | `2` |
| `194550` | `1` |
| `194551` | `1,4` |

> Test 2

Enter `Focus Run #:` 194747
The focused file should still be imported on the right hand side of the interface within the list view widget but should be disabled
The bank files are still loaded correctly

> Other Tests

If one of the run file is missing, an appropriate warning should be displayed. 
Invalid run number range is handled appropriately
Other features still function as they should
Check the `EnggDiffractionPresenterTest` passes
Documentation has been updated



Fixes #15906 .
[Release Notes](https://github.com/mantidproject/mantid/blob/d0cd18a5f64db536eeaec47d1278f6d88acc4298/docs/source/release/v3.7.0/diffraction.rst)



---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
